src/Symfony/Component/HttpKernel/EventListener/ExceptionListener.php

```diff
@@ -49,6 +49,15 @@ public function __construct($controller, LoggerInterface $logger = null, $debug
 
     public function onKernelException(GetResponseForExceptionEvent $event)
     {
+        while (null === $this->controller) {
```

Comment:
should be a `if` rather than a `while`. It would be much easier to understand, and it can never loop anyway.

Reply:
using `while` allows using the `break` below :)

Reply:
while removed btw :)